common/internal_model/src/main/scala/weco/catalogue/internal_model/work/AbstractRootConcept.scala

```diff
@@ -6,11 +6,36 @@ sealed trait AbstractRootConcept[+State] extends HasId[State] {
   val label: String
 }
 
+object AbstractRootConcept {
+
+  private val conceptTypeMap
+    : Map[String,
+          (IdState.Identifiable, String) => AbstractRootConcept[
+            IdState.Identifiable
+          ]] = Map(
+    "Concept" -> (new Concept[IdState.Identifiable](_, _)),
+    "Agent" -> (new Agent[IdState.Identifiable](_, _)),
+    "Place" -> (new Place[IdState.Identifiable](_, _)),
+    "Period" -> (new Period[IdState.Identifiable](_, _)),
+    "Meeting" -> (new Meeting[IdState.Identifiable](_, _)),
+    "Organisation" -> (new Organisation[IdState.Identifiable](_, _)),
+    "Person" -> (new Person[IdState.Identifiable](_, _))
+  )
```

Comment:
This is annoying isn't it - is there any way round it?

Reply:
I wondered about an enum, but I think that's no less annoying than this.

Strictly, none of this is necessary.  At this point, I don't think there is a difference in the behaviour of any of these types (there is when constructing them from parsed MARC, but if we reach the point where we are changing one type for another, it will not have the extra properties associated with the new type).

However, downstream of here, (in the display format stuff), they derive the type value in the JSON from the Scala type.